com.unity.render-pipelines.universal/ShaderLibrary/Lighting.hlsl

```diff
@@ -627,21 +627,165 @@ half3 SampleLightmap(float2 lightmapUV, half3 normalWS)
+    bool probe1Dominant = importanceSign < 0 || (importanceSign == 0 && volumeSign > 0);
+
+    float desiredWeightProbe0 = CalculateProbeWeight(positionWS, unity_SpecCube0_BoxMin, unity_SpecCube0_BoxMax);
+    float desiredWeightProbe1 = CalculateProbeWeight(positionWS, unity_SpecCube1_BoxMin, unity_SpecCube1_BoxMax);
+
+    // Subject the probes weight if the other probe is dominant
+    float weightProbe0 = probe1Dominant ? min(desiredWeightProbe0, 1 - desiredWeightProbe1) : desiredWeightProbe0;
+    float weightProbe1 = probe0Dominant ? min(desiredWeightProbe1, 1 - desiredWeightProbe0) : desiredWeightProbe1;
+
+    float totalWeight = weightProbe0 + weightProbe1;
+
+    // If either probe 0 or probe 1 is dominant the sum of weights is guaranteed to be 1.
+    // If neither is dominant this is not guaranteed - only normalize weights if totalweight exceeds 1.
+    weightProbe0 /= max(totalWeight, 1);
+    weightProbe1 /= max(totalWeight, 1);
+
+    half3 irradiance = half3(0, 0, 0);
+    half3 originalReflectVector = reflectVector;
+    half mip = PerceptualRoughnessToMipmapLevel(perceptualRoughness);
+
+    // Sample the first reflection probe
+    if (weightProbe0 > 0.01)
+    {
+#ifdef _REFLECTION_PROBE_BOX_PROJECTION
+        reflectVector = BoxProjectedCubemapDirection(originalReflectVector, positionWS, unity_SpecCube0_ProbePosition, unity_SpecCube0_BoxMin, unity_SpecCube0_BoxMax);
+#else
+        reflectVector = originalReflectVector;
+#endif // _REFLECTION_PROBE_BOX_PROJECTION
+
+        half4 encodedIrradiance = half4(SAMPLE_TEXTURECUBE_LOD(unity_SpecCube0, samplerunity_SpecCube0, reflectVector, mip));
+
+#if defined(UNITY_USE_NATIVE_HDR) || defined(UNITY_DOTS_INSTANCING_ENABLED)
+        irradiance += weightProbe0 * encodedIrradiance.rbg;
+#else
+        irradiance += weightProbe0 * DecodeHDREnvironment(encodedIrradiance, unity_SpecCube0_HDR);
+#endif // UNITY_USE_NATIVE_HDR || UNITY_DOTS_INSTANCING_ENABLED
+    }
+
+    // Sample the second reflection probe
+    if (weightProbe1 > 0.01)
+    {
+#ifdef _REFLECTION_PROBE_BOX_PROJECTION
+        reflectVector = BoxProjectedCubemapDirection(originalReflectVector, positionWS, unity_SpecCube1_ProbePosition, unity_SpecCube1_BoxMin, unity_SpecCube1_BoxMax);
+#else
+        reflectVector = originalReflectVector;
+#endif // _REFLECTION_PROBE_BOX_PROJECTION
+        half4 encodedIrradiance = half4(SAMPLE_TEXTURECUBE_LOD(unity_SpecCube1, samplerunity_SpecCube1, reflectVector, mip));
+
+#if defined(UNITY_USE_NATIVE_HDR) || defined(UNITY_DOTS_INSTANCING_ENABLED)
+        irradiance += weightProbe1 * encodedIrradiance.rbg;
+#else
+        irradiance += weightProbe1 * DecodeHDREnvironment(encodedIrradiance, unity_SpecCube1_HDR);
+#endif // UNITY_USE_NATIVE_HDR || UNITY_DOTS_INSTANCING_ENABLED
+    }
+
+    // Use any remaining weight to blend to environment reflection cube map
+    if (totalWeight < 0.99)
+    {
+        reflectVector = originalReflectVector;
```

Comment:
Instead of doing assignment, just pass in original copy.

Reply:
resolved in 203240f